tests/test_ote_training.py

```diff
@@ -0,0 +1,690 @@
+from sc_sdk.entities.dataset_storage import NullDatasetStorage
+
+from e2e_test_system import e2e_pytest_performance, DataCollector
+
+
+#logger_name = osp.splitext(osp.basename(__file__))[0]
+#logger = logger_factory.get_logger(logger_name)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+def DATASET_PARAMETERS_FIELDS():
+    return ['annotations_train',
+            'images_train_dir',
+            'annotations_val',
+            'images_val_dir',
+            'annotations_test',
+            'images_test_dir',
+            ]
+
+ROOT_PATH_KEY = '_root_path'
+DatasetParameters = namedtuple('DatasetParameters', DATASET_PARAMETERS_FIELDS())
+
+@pytest.fixture
+def dataset_definitions_fx(request):
+    """
+    Return dataset definitions read from a YAML file passed as the parameter --dataset-definitions.
+    Note that the dataset definitions should store the following structure:
+    {
+        <dataset_name>: {
+            'annotations_train': <annotation_file_path1>
+            'images_train_dir': <images_folder_path1>
+            'annotations_val': <annotation_file_path2>
+            'images_val_dir': <images_folder_path2>
+            'annotations_test': <annotation_file_path3>
+            'images_test_dir':  <images_folder_path3>
+        }
+    }
+    """
+    path = request.config.getoption('--dataset-definitions')
+    assert path is not None, (f'The command line parameter "--dataset-definitions" is not set, '
+                             f'whereas it is required for the test {request.node.originalname or request.node.name}')
+    with open(path) as f:
+        data = yaml.safe_load(f)
+    data[ROOT_PATH_KEY] = osp.dirname(path)
+    return data
+
+@pytest.fixture
+def template_paths_fx(request):
+    """
+    Return mapping model names to template paths, received from globbing the folder configs/ote/
+    Note that the function searches files with name `template.yaml`, and for each such file
+    the model name is the name of the parent folder of the file.
+    """
+    root = osp.dirname(osp.dirname(osp.realpath(__file__)))
+    glb = glob.glob(f'{root}/configs/ote/**/template.yaml', recursive=True)
+    data = {}
+    for p in glb:
+        assert osp.isabs(p), f'Error: not absolute path {p}'
+        name = osp.basename(osp.dirname(p))
+        if name in data:
+            raise RuntimeError(f'Duplication of names in config/ote/ folder: {data[name]} and {p}')
+        data[name] = p
+    data[ROOT_PATH_KEY] = ''
+    return data
+
+def _make_path_be_abs(some_val, root_path):
+    assert isinstance(some_val, (str, dict)), f'Wrong type of value: {some_val}, type={type(some_val)}'
+    assert isinstance(root_path, str), f'Wrong type of root_path: {root_path}, type={type(root_path)}'
+
+    if isinstance(some_val, str):
+        if not osp.isabs(some_val):
+            return osp.join(root_path, some_val)
+        return some_val
+
+    some_dict = some_val
+    for k in sorted(some_dict.keys()):
```

Comment:
why do you need to sort it?

Reply:
We cannot iterate through `dict_keys` object, returned by `.keys` method, if we change the dict.
Ok, I will make `list(...)` instead of `sorted(...)`